### DISCUSSION TITLE
What's the state of `useProxy`?

### DISCUSSION THREAD

**Discussion by piscopancer** [pmndrs/valtio#832] [General]
Hey, I have spent several hours reading issues and discussions while trying to accumulate as much information as my brain can handle to figure out if I want to try out useProxy. As the result, I have not decided yet. And as I noticed an unsettling amount of people being unsatisfied with the separation of read/write, where read is useSnapshot, obviously, I wanted to ask:
- if useProxy destructuring still creates snapshots, which is ahh, unwanted, I assume;
- if there is anything else that useProxy does wrong or poorly, that needs to be kept in mind while using it.

Also, I was excited to use proxyMap from utils module, however I ran into webpack issues in my Next app while using .get() function. And the reason why I justified using map instead of array is not performance (like the majority say), but rather the fact that I need key+value entity. So I wanted to know, if I am recommended to use proxy object, not map, as I am not driven by benefiting from map's performance (array sucks in my scenario because all entries are unique and have ids)

Kiss 💋

**Comment by dai-shi** [comment]
> What's the state of useProxy?

Sorry for the confusion. `useProxy` util is pretty stable in terms of its implementation. What's poor is docs. We should probably show do's and dont's examples more clearly.

(If anyone is interested in contributing. Please let me know.)

**Comment by dai-shi** [comment]
> if useProxy destructuring still creates snapshots

It does. It's still the same as useSnapshot.

If you mean, you don't want the auto-tracking snapshots, you can manually subscribe to valtio proxies:
https://github.com/pmndrs/valtio/blob/58f17ba0113a0dd080f69e6ac1f323cd0ba5886a/docs/how-tos/how-to-avoid-rerenders-manually.mdx#L29

**Comment by dai-shi** [comment]
> the reason why I justified using map instead of array is not performance

Note that proxyMap from valtio/utils is not for performance. It's just an emulation using plain objects. So, if you can use plain objects instead, it would be simpler and more predictable.
